pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -283,25 +293,54 @@ func (manager *Manager) cleanCluster(name string) error {
 	return manager.storage.CleanCluster(context.TODO(), name)
 }
 
-func (manager *Manager) UpdateClusterStatus(ctx context.Context, name string, status *clusterv1alpha2.ClusterStatus) error {
-	cluster, err := manager.clusterlister.Get(name)
-	if err != nil {
-		return err
+func (manager *Manager) UpdateClusterSynchroCondition(name, reason, message string, status metav1.ConditionStatus) {
+	synchroCondition := metav1.Condition{
+		Type:    clusterv1alpha2.ClusterSynchroCondition,
+		Reason:  reason,
+		Status:  status,
+		Message: message,
 	}
 
-	if equality.Semantic.DeepEqual(&cluster.Status, status) {
-		return nil
-	}
+	clusterStatus := &clusterv1alpha2.ClusterStatus{Conditions: []metav1.Condition{synchroCondition}}
 
-	cluster = cluster.DeepCopy()
-	cluster.Status = *status
-	_, err = manager.clusterpediaclient.ClusterV1alpha2().PediaClusters().UpdateStatus(ctx, cluster, metav1.UpdateOptions{})
-	if err != nil {
-		return err
+	if err := manager.UpdateClusterStatus(context.TODO(), name, clusterStatus); err != nil {
+		klog.ErrorS(err, "Failed to update cluster status", "cluster", name, clusterStatus.Conditions[0].Reason)
 	}
+}
 
-	klog.V(2).InfoS("Update Cluster Status", "cluster", cluster.Name, "status", status.Conditions[0].Reason)
-	return nil
+func (manager *Manager) UpdateClusterStatus(ctx context.Context, name string, status *clusterv1alpha2.ClusterStatus) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cluster, err := manager.clusterlister.Get(name)
+		if err != nil {
+			return err
+		}
+
+		if equality.Semantic.DeepEqual(&cluster.Status, status) {
+			return nil
+		}
+
+		cluster = cluster.DeepCopy()
+
+		currentClusterStatus := cluster.Status
+		if status.Version != "" {
+			currentClusterStatus.Version = status.Version
+		}
+		if status.SyncResources != nil {
+			currentClusterStatus.SyncResources = status.SyncResources
+		}
+		for _, condition := range status.Conditions {
+			meta.SetStatusCondition(&currentClusterStatus.Conditions, condition)
+		}
+
+		cluster.Status = currentClusterStatus
+
+		_, err = manager.clusterpediaclient.ClusterV1alpha2().PediaClusters().UpdateStatus(ctx, cluster, metav1.UpdateOptions{})
+		if err == nil {
+			klog.V(2).InfoS("Update Cluster Status", "cluster", cluster.Name, "type", status.Conditions[0].Type, "status", status.Conditions[0].Status, "reason", status.Conditions[0].Reason, "message", status.Conditions[0].Message)
```

Comment:
It may be necessary to show the ReadyCondition information as well.
I think it would be better for the caller of  `UpdateClusterStatus` to print the log, since they know which state may have been modified.

This can be fixed separately in other pr.

Reply:
ok